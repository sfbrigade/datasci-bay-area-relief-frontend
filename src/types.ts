export enum County {
    SanFrancisco = 'San Francisco',
    Alameda = 'Alameda',
    SanMateo = 'San Mateo',
    ContraCosta = 'Contra Costa',
    SantaClara = 'Santa Clara',
    Any = 'Any'
}

export enum BusinessType {
    SmallBusiness = 'Small business',
    NonProfit = 'Non-profit'
}

export interface FilterOptions {
    businessType: BusinessType;
    county: County;
}

export interface Result {
    name: string;
}
