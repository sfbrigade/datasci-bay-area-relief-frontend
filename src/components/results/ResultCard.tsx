import React from "react";
import Moment from "moment";

import {Result} from "../../types";
import {
  formatAwardAmount,
  formatDate,
  formatInterestRate,
  formatReliefType,
} from "./formatHelpers";
import Button from "@mui/material/Button";
import {
  CardBottom,
  StyledAwardAmount,
  StyledCard,
  StyledName,
  StyledReliefType,
  SupportTypeContainer,
  SupportTypeItem, Tag, Tags, ExpiredTag
} from "./ResultCard.styles";

const onApply = (url: string) => {
  window.open(url, "_blank");
};

const ResultCard: React.FC<Result> = ({
  supportType,
  interestRate,
  deadline,
  deadlineApplicable,
  maxAwardAmount,
  name,
  reliefType,
  blackOwned,
  lgbtq,
  websiteUrl,
}) => {
  let isExpired = false;
  let deadlineLabelText = "";
  if (deadlineApplicable === "Yes" || deadlineApplicable === "Unknown") {
    if (!deadline) {
      deadlineLabelText = "Unknown";
    } else {
      deadlineLabelText = formatDate(deadline);
      isExpired = Moment().isAfter(Moment(deadline));
    }
  } else if (deadlineApplicable === "No") {
    deadlineLabelText = "Ongoing";
  }

  return (
    <StyledCard>
      <SupportTypeContainer>
        <SupportTypeItem>
          <span data-testid='support-type-header'>{`${supportType} • ${formatInterestRate(
            interestRate,
            supportType
          )} • ${deadlineLabelText}`}</span>
        </SupportTypeItem>
      </SupportTypeContainer>
      <StyledAwardAmount>{formatAwardAmount(maxAwardAmount)}</StyledAwardAmount>
      <StyledName>{name}</StyledName>
      <CardBottom>
        {websiteUrl !== "None" && (
          <Button color="secondary" onClick={() => onApply(websiteUrl)}>
            Apply
          </Button>
        )}

        <Tags>
          {blackOwned && <Tag>Black-owned</Tag>}
          {lgbtq && <Tag>LGBTQ</Tag>}
          {isExpired && <ExpiredTag>Possibly Closed</ExpiredTag>}
        </Tags>
      </CardBottom>
    </StyledCard>
  );
};

export default ResultCard;
