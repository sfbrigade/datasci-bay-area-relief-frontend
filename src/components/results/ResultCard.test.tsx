import React from "react";
import ResultCard from "./ResultCard";
import renderer from "react-test-renderer";
import {ReliefType, SupportType} from "../../types";

describe('ResultCard tests', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<ResultCard
        id={1}
        name="some test name"
        supportType={SupportType.Grant}
        interestRate={0.01}
        dateAdded={"Wed, 10 Jun 2020 00:00:00 PST"}
        maxAwardAmount={10000}
        reliefType={ReliefType.COVID}
        deadline={"Mon, 15 Jun 2020 00:00:00 PST"}
        deadlineApplicable={"Yes"}
        smallBusiness={true}
        nonProfit={false}
        employs100OrFewer={false}
        employs500OrFewer={false}
        employs750OrFewer={false}
        employs750More={true}
        sanMateoCounty={false}
        contraCostaCounty={false}
        santaClaraCounty={false}
        sfCounty={false}
        alamedaCounty={true}
        hasInterest={true}
        doesNotHaveInterest={false}
        covid19={true}
        protestDamage={false}
        blackOwned={true}
        lgbtq={false}
        public={false}
        private={true}
        spanish={false}
        chinese={false}
        websiteUrl="https://google.com" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
 });
});
