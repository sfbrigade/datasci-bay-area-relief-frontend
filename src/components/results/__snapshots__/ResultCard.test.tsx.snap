// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard tests matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root sc-bdnylx cJtASr MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="sc-gtssRu frKVdL"
  >
    <span
      className="MuiTypography-root sc-dlnjPT gBKAmh MuiTypography-overline"
    >
      Grant • No Interest • 6/10/20
    </span>
  </div>
  <h5
    className="MuiTypography-root sc-hKFyIo gDTYSf MuiTypography-h5"
  >
    $10,000
  </h5>
  <p
    className="MuiTypography-root sc-eCApGN hgNLgN MuiTypography-body2"
  >
    some test name
  </p>
  <p
    className="MuiTypography-root sc-jSFkmK jvsbkj MuiTypography-body2"
  >
    COVID relief
  </p>
  <div
    className="sc-gKAblj fKEQtp"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Apply
      </span>
    </button>
    <div
      className="sc-iCoHVE jCbugA"
    >
      <p
        className="MuiTypography-root sc-fujyUd fyCOjw MuiTypography-body2"
      >
        Black-owned
      </p>
    </div>
  </div>
</div>
`;
